### PR TITLE
New reactions

### PR DESCRIPTION
**Pull request by bago2k4**
Card: https://trello.com/c/D3eMyR2a
Abstract: https://share.goabstract.com/c5132b02-d011-4291-bc1f-fbc150ff7002

To test:
- pick a reaction
- react with a fav reaction
- unreact a reacted
- react/unreact without refreshing the page
- any error? NO? good!

At the moment the slot animation is missing, i'll open another PR for it

### CODE REVIEW

**Comment by bago2k4** [comment]
[Jelly UX: New reactions UI](https://trello.com/c/D3eMyR2a/1184-jelly-ux-new-reactions-ui)

**Comment by belucid** [comment]
Let's see what the desired favorites treatment on this one is before we roll it out.

File: src/oc/web/actions/reaction.cljs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            [oc.web.lib.jwt :as jwt]
+            [oc.web.router :as router]
+            [oc.web.dispatcher :as dis]
+            [oc.web.lib.utils :as utils]
+            [oc.web.lib.json :refer (json->cljs)]))
+
+(defn react-from-picker [activity-data emoji]
+  (dis/dispatch! [:handle-reaction-to-entry activity-data
+   {:reaction emoji :count 1 :reacted true :links [] :authors []}])
+  (when (and emoji
```
Comment (thepug) [inline src/oc/web/actions/reaction.cljs]: is there a case where the emoji is nil/false?

File: src/oc/web/actions/reaction.cljs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+             (utils/link-for (:links activity-data) "react"))
+    (api/react-from-picker activity-data emoji
+      (fn [{:keys [status success body]}]
+        (api/get-entry activity-data)
```
Comment (thepug) [inline src/oc/web/actions/reaction.cljs]: leave a comment here for why the `get-entry` call is needed.

File: src/oc/web/stores/reaction.cljs [hunk 3]
```diff
@@ -0,0 +1,150 @@
+(ns oc.web.stores.reaction
+  (:require [taoensso.timbre :as timbre]
+            [oc.web.lib.jwt :as jwt]
+            [oc.web.router :as router]
+            [oc.web.lib.utils :as utils]
+            [oc.web.dispatcher :as dispatcher]))
+
+(defn handle-reaction-to-entry-finish
+  [db activity-data reaction reaction-data]
+  (let [board-data (get-in db (dispatcher/current-board-key))
```
Comment (thepug) [inline src/oc/web/stores/reaction.cljs]: board-data is unused in this function

File: src/oc/web/stores/reaction.cljs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                                    (assoc reactions-data reaction-idx next-reaction-data))
+                                  (vec (remove #(= (:reaction %) (name reaction)) reactions-data)))
+            updated-activity-data (-> activity-data
+                                   (assoc :reactions-loading next-reactions-loading)
+                                   (assoc :reactions next-reactions-data))]
+        (assoc-in db activity-key updated-activity-data)))))
+
+(defn handle-reaction-to-entry [db activity-data reaction-data]
+  (let [board-key (dispatcher/current-board-key)
+        board-data (get-in db board-key)
```
Comment (thepug) [inline src/oc/web/stores/reaction.cljs]: board-data is unused in this function

File: src/oc/web/actions/reaction.cljs (hunk 1 above)
Comment (bago2k4) [reply src/oc/web/actions/reaction.cljs]: The point is that we need to use native emojis, the picker pass to use a js object containing not only the emoji char but some information about it and some times `emoji.native` can be null.
That's why i added that check, adding a comment.

**Comment by bago2k4** [comment]
@thepug cool, fixed all.. ready again for review.

**Pull request merged**